### PR TITLE
libvirt_vm: use utils_package for package installation

### PR DESCRIPTION
**Pull request by balamuruhans**
reuse utils_package for install/remove packages and remove existing
redundant code.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/libvirt_vm.py [hunk 1]
```diff
@@ -2887,10 +2812,10 @@ def prepare_guest_agent(self, prepare_xml=True, channel=True, start=True):
         if not self.is_alive():
             self.start()
 
-        self.install_package('pm-utils', ignore_status=True)
-        self.install_package('qemu-guest-agent')
-
         session = self.wait_for_login()
+        packages = ['pm-utils', 'qemu-guest-agent']
```
Comment (sathnaga) [inline virttest/libvirt_vm.py]: we can keep `vm.install_package()` method as is and just use `utils_package.package_install(packages, session) ` inside that method, will help us in backward compatibility aswell and we can avoid taking session in all usage places...

Comment (balamuruhans) [reply virttest/libvirt_vm.py]: @sathnaga I agree and it is the better way. Thanks you!

File: virttest/libvirt_vm.py [hunk 2]
```diff
@@ -2795,73 +2796,28 @@ def install_package(self, name, ignore_status=False):
...
-                if status:
-                    if not ignore_status:
-                        raise virt_vm.VMError("Update of package %s failed:"
-                                              "\n%s" % (name, output))
-                    else:
-                        logging.error("Update of package %s failed:"
-                                      "\n%s", name, output)
-        finally:
-            session.close()
+        if not utils_package.package_install(name, session):
```
Comment (chunfuwen) [inline virttest/libvirt_vm.py]: suggest add one check whether session is None before call it in utils_package.package_install

File: virttest/libvirt_vm.py [hunk 3]
```diff
@@ -2795,73 +2796,28 @@ def install_package(self, name, ignore_status=False):
...
+            if not ignore_status:
+                session.close()
```
Comment (chunfuwen) [inline virttest/libvirt_vm.py]: session.close() call twice in line 2801 and 2805,actually can merge call once if put it before line 2800

File: virttest/libvirt_vm.py (hunk 2 above)
Comment (balamuruhans) [reply virttest/libvirt_vm.py]: I believe wait_for_login will raise error if it fails and checking session is not necessary here

File: virttest/libvirt_vm.py (hunk 3 above)
Comment (balamuruhans) [reply virttest/libvirt_vm.py]: we cant keep session.close() in 2800, then session will close only on utils_package fails. I think it is in right place.

**Comment by hsmj1412** [comment]
There are some test error when to destroy the guest, because the "virsh destroy" operation will damage the guest repo's database. So we need try to clean the repo and re-try to install.

**Pull request merged**